app/src/androidTest/java/com/github/se/cyrcle/model/parking/OfflineParkingRepositoryRoomTest.kt

```diff
@@ -0,0 +1,230 @@
+package com.github.se.cyrcle.model.parking
+
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.se.cyrcle.model.parking.offline.OfflineParkingRepositoryRoom
+import com.github.se.cyrcle.model.parking.offline.ParkingDao
+import com.github.se.cyrcle.model.parking.offline.ParkingDatabase
+import com.github.se.cyrcle.model.parking.offline.UnSupportedOperationException
+import java.util.concurrent.CountDownLatch
+import kotlinx.coroutines.test.runTest
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertThrows
+import org.junit.Assert.fail
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class OfflineParkingRepositoryRoomTest {
+
+  private lateinit var database: ParkingDatabase
+  private lateinit var parkingDao: ParkingDao
+  private lateinit var repository: OfflineParkingRepositoryRoom
+
+  private val testList =
+      listOf(
+          TestInstancesParking.parking1,
+          TestInstancesParking.parking2,
+          TestInstancesParking.parking3,
+          TestInstancesParking.parking4,
+          TestInstancesParking.parking5)
+
+  @Before
+  fun setUp() = runTest {
+    database =
+        Room.inMemoryDatabaseBuilder(
+                ApplicationProvider.getApplicationContext(), ParkingDatabase::class.java)
+            .allowMainThreadQueries()
+            .build()
+    parkingDao = database.parkingDao
+    repository = OfflineParkingRepositoryRoom(database)
+
+    parkingDao.upsert(TestInstancesParking.parking1)
+  }
+
+  @After
+  fun tearDown() {
+    database.close()
```

Comment:
*[TEMPORARY COMMENT WHILE ANDROID STUDIO IS SYNCING]*
Wouldn't it be cleaner to also destroy the database as well ? I feel like this can be some bug source if we don't delete the database between tests. I'm ready to believe that JUnit recreates the whole environment each time, but I'm not an expert on this subject.

Reply:
Mhhhh I'm not to sure. I think that because the `@Before` creates a new database, it ovewrites the database already created and held in that reference. Since those test databases are in memory, they will simply be deleted by the garbage collector?

Reply:
Would you be willing to test both and report back? This could be a MAJOR error source so I'd be grateful if you tried before merging the PR. I am however not an expert either, so if you're confident I'll take your word for it